src/cmd/ksh93/bltins/math.c

```diff
@@ -100,13 +100,17 @@ static Sfdouble_t local_nexttoward(int type_1, Sfdouble_t arg_1, int type_2, Sfd
 
     switch (type_1) {
         case 1: {
-            return nexttowardf((float)arg_1, arg_2);
+            return nexttowardf((float)arg_1, (float)arg_2);
```

Comment:
None of these explicit casts are required. And while you can argue they provide useful information to a reader of the code they can also cause bugs. What if `nexttowardf()` actually took a different type such as a `double`. This will compile without warning but will cause two conversions and lost of information.